infrastructure/edgeRedirectLambdaAssociations.ts

```diff
@@ -0,0 +1,124 @@
+                        headers: {
+                            "location": [
+                                {
+                                    key: "Location",
+                                    value: redirect,
+                                },
+                            ],
+                            "cache-control": [
+                                {
+                                    key: "Cache-Control",
+                                    value: "max-age=3600",
+                                },
+                            ],
+                        },
+                    };
+
+                    // Use a modified response.
+                    callback(null, modifiedResponse);
+
+                } else {
+
+                    // Use the original request.
+                    callback(null, request)
+                }
+            },
+            funcDescription: "Lambda function that conditionally redirects based on a path-matching expression.",
+        },
+        {}
+    );
+
+    return [
+        {
+            includeBody: false,
+            lambdaArn: lambdaEdge.getLambdaEdgeArn(),
+            eventType: "origin-request",
+        }
+    ];
+}
+
+function getRedirect(uri: string): string | undefined {
+    return nodeSDKRedirect(uri) || pythonSDKRedirect(uri) || dotnetSDKRedirect(uri) || undefined;
+}
+
+function nodeSDKRedirect(uri: string): string | undefined {
+    const pattern = new URLPattern("/docs/reference/pkg/nodejs/pulumi/:provider(/:service)(*)");
+    const match = pattern.match(uri);
+    const exceptions = [
+        "pulumi",
+        "cloud",
+        "policy",
+        "awsx",
+        "eks",
+        "kubernetesx",
+        "terraform",
+    ];
+
+    if (match && match.provider && !exceptions.includes(match.provider)) {
+        if (match.service) {
+            return `/docs/reference/pkg/${match.provider}/${match.service}/?language=nodejs`;
+        }
+        return `/docs/reference/pkg/${match.provider}/?language=nodejs`;
+    }
+
+    return undefined;
+};
+
+function pythonSDKRedirect(uri: string): string | undefined {
+    const pattern = new URLPattern("/docs/reference/pkg/python/pulumi_(:provider)(/:service)(*)");
+    const match = pattern.match(uri);
+    const exceptions = [
+        "pulumi",
+        "policy",
+        "terraform",
+    ];
+
+    if (match && match.provider && !exceptions.includes(match.provider)) {
+        if (match.service) {
+            return `/docs/reference/pkg/${match.provider}/${match.service}/?language=python`;
+        }
+        return `/docs/reference/pkg/${match.provider}/?language=python`;
+    }
+
+    return undefined;
+};
+
+function dotnetSDKRedirect(uri: string): string | undefined {
+    const pattern = new URLPattern("/docs/reference/pkg/dotnet/Pulumi.(:provider)/Pulumi.(:providerAgain)(.(:service)).html");
+    const match = pattern.match(uri);
+    const exceptions = [
+        "Pulumi",
+        "Policy",
+        "Terraform",
```

Comment:
Actually, this isn't right, as these don't exist. I'll have to direct them to GitHub instead, I think.